BLE Indoor Positioning/src/test/java/com/nexenio/bleindoorpositioning/location/provider/DeviceLocationPredictorTest.java

```diff
@@ -0,0 +1,47 @@
+package com.nexenio.bleindoorpositioning.location.provider;
+
+import com.nexenio.bleindoorpositioning.location.Location;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.nexenio.bleindoorpositioning.location.provider.DeviceLocationPredictor.predictLocationFromLocations;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by leon on 14.02.18.
+ */
+public class DeviceLocationPredictorTest {
+
+    public static final double HUMAN_WALKING_SPEED = 1.388889; // meters per second
+
+    public static final Location SQUARE_ONE_GENDARMENMARKT = new Location(52.513588, 13.392995);
+    public static final Location SQUARE_TWO_GENDARMENMARKT = new Location(52.513632, 13.392986);
+    public static final Location SQUARE_THREE_GENDARMENMARKT = new Location(52.513675, 13.392979);
+    public static final Location SQUARE_FOUR_GENDARMENMARKT = new Location(52.513720, 13.392971);
+    public static final Location SQUARE_EXPECTED_GENDARMENMARKT = new Location(52.513764, 13.392962);
+
+    @Test
+    public void predictLocationFromLocations_locations_accuratePrediction() throws Exception {
+        List<Location> strollOnGendarmenmarkt = new ArrayList<>();
+        strollOnGendarmenmarkt.add(SQUARE_ONE_GENDARMENMARKT);
+        strollOnGendarmenmarkt.add(SQUARE_TWO_GENDARMENMARKT);
+        strollOnGendarmenmarkt.add(SQUARE_THREE_GENDARMENMARKT);
+        strollOnGendarmenmarkt.add(SQUARE_FOUR_GENDARMENMARKT);
```

Comment:
This belongs to a setup method. Also directly set the `lastChangeTimestamp` as it will be used in other tests as well.